components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1865,11 +1862,11 @@ void AdsServiceImpl::OnPrefsChanged(const std::string& pref) {
     StartCheckIdleStateTimer();
   } else if (pref == brave_rewards::prefs::kWalletBrave) {
     OnWalletUpdated();
-#if BUILDFLAG(ENABLE_BRAVE_NEWS)
-  } else if (pref == brave_news::prefs::kBraveTodayOptedIn ||
-             pref == brave_news::prefs::kNewTabPageShowToday) {
+  } else if (base::FeatureList::IsEnabled(
```

Comment:
If the Brave news feature is disabled, would it be possible for a user to toggle the switches which trigger a pref change?

Reply:
not sure if i understand correctly, but it the option to enable/disable news should still appear in brave://flags

Reply:
`OnPrefsChanged` would only get called if a pref changes. And I assume the only way a pref can change is if the user enables/opts in to Brave News. Therefore surely the `base::FeatureList::IsEnabled(brave_today::features::kBraveNewsFeature)` condition can be removed?

Reply:
@tmancey actually right now the sources can be accessed even when show news is disabled.  i will fix that afterwards as part of improvements and i will remove this condition too.

Reply:
I understand that the sources can be accessed. But, if this feature is not enabled via `brave_today::features::kBraveNewsFeature` then these prefs should never be triggered as the switches would not be visible in the UI

Reply:
@tmancey not sure about that. i would say no. @petemill wdyt?
also removed condition.